cpp/src/arrow/array/array_nested.h

```diff
@@ -125,13 +125,19 @@ class ARROW_EXPORT ListArray : public BaseListArray<ListType> {
   /// \param[in] values Array containing list values
   /// \param[in] pool MemoryPool in case new offsets array needs to be
   /// allocated because of null values
+  /// \param[in] null_bitmap Array of valid and null values,
+  /// where 0 represents null and 1 represents valid.
+  /// \param[in] null_count Count of null values
   static Result<std::shared_ptr<ListArray>> FromArrays(
-      const Array& offsets, const Array& values,
-      MemoryPool* pool = default_memory_pool());
+      const Array& offsets, const Array& values, MemoryPool* pool = default_memory_pool(),
+      std::shared_ptr<Buffer> null_bitmap = NULLPTR,
+      int64_t null_count = kUnknownNullCount);
 
   static Result<std::shared_ptr<ListArray>> FromArrays(
       std::shared_ptr<DataType> type, const Array& offsets, const Array& values,
-      MemoryPool* pool = default_memory_pool());
+      MemoryPool* pool = default_memory_pool(),
+      std::shared_ptr<Buffer> null_bitmap = NULLPTR,
+      int64_t null_count = kUnknownNullCount);
 
```

Comment:
Also, should similar changes be done for `LargeListArray`?

Reply:
Makes sense, couldn't get the unit tests in cpp to pass without it anyway. :) https://github.com/apache/arrow/pull/13894/commits/819f0c17e5cf412c944165d0a8b2b495967f0970